### ISSUE TITLE
tzdata 2020c update breaks some applications

### ISSUE THREAD

**Issue by buckley310** [NixOS/nixpkgs#103317]
**Describe the bug**
The update from tzdata 2019c to 2020c results in the GNOME panel and AwesomeWM panel displaying the incorrect time.
In my case, the displayed time represents daylight savings time, even though daylight savings has ended, and settings displays EDT instead of EST as the timezone.

**To Reproduce**
1. Be on nixos-unstable
2. Use GNOME or Awesome Window Manager

**Additional context**
Rebuilding on a52e974cff8 makes the issue appear. (this is the latest revision of nixos-unstable as of writing)
When I checked out a52e974cff8, reverted 8904ce243aa885e0295610bd4e1dc7ca293e8919, and rebuilt on that locally, the issue was no longer present.
There may be other software affected, but these are the only ones I have seen.
The `date` command and some other software on my system that use the system time still display the correct time in all cases.

**Notify maintainers**

@fpletz
@ajs124 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.75, NixOS, 21.03pre250878.a52e974cff8 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(sean): `""`
 - channels(root): `"sconfig, nixos-21.03pre250878.a52e974cff8"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
    tzdata
```

**Comment by felschr** [comment]
tzdata `2020d` has been released meanwhile, perhaps they fixed the issue.

**Comment by ajs124** [comment]
This is/was probably fixed by #102946

**Comment by jtojnar** [comment]
Yes, this should be partially fixed by #102946. There was also another regression in GLib which has been fixed in 30b932e46b13a9ad6640c0429c97c2af20458228.

**Comment by buckley310** [comment]
Yep, looks like rebuilding directly from master fixes the issue, at least for AWM (I don't feel like compiling all of gnome right now).
Perhaps I should have tried that first.
Thanks for the replies. I will close this on the assumption the fixes will hit the channel soon.
